tests/FSharp.Test.Utilities/Compiler.fs

```diff
@@ -0,0 +1,497 @@
+open System.Collections.Immutable
+open System.IO
+
+module Compiler =
+
+    type TestType =
+        | Text of string
+        | Path of string
+        | Baseline of (string * string)
+
+    type CompilationUnit =
+        | FS  of FSharpCompilationSource
+        | CS  of CSharpCompilationSource
+        | IL  of ILCompilationSource
+
+    and FSharpCompilationSource =
+        { Source:         TestType
+          Options:        string list
+          OutputType:     CompileOutput
+          SourceKind:     SourceKind
+          Name:           string option
+          IgnoreWarnings: bool
+          References:     CompilationUnit list }
+
+    and CSharpCompilationSource =
+        { Source:          TestType
+          LangVersion:     CSharpLanguageVersion
+          TargetFramework: TargetFramework
+          Name:            string option
+          References:      CompilationUnit list }
+
+    and ILCompilationSource =
+        { Source:     TestType
+          References: CompilationUnit list}
+
+    type ErrorType = Error of int | Warning of int
+
+    type Line = Line of int
+    type Col = Col of int
+
+    type Range = { StartLine:   int
+                   StartColumn: int
+                   EndLine:     int
+                   EndColumn:   int }
+
+    type ErrorInfo = { Error:   ErrorType
+                       Range:   Range
+                       Message: string }
+
+    type Output = { OutputPath: string option
+                    Adjust:     int
+                    Errors:     ErrorInfo list
+                    Warnings:   ErrorInfo list }
+
+    type CompilationResult =
+        | Success of Output
+        | Failure of Output
+
+    let private defaultOptions : string list = []
+
+    // Not very safe version of reading stuff from file, but we want to fail fast for now if anything goes wrong.
+    let private getSource (src: TestType) : string =
+        match src with
+        | Text t -> t
+        | Path p -> System.IO.File.ReadAllText p
+        | Baseline (d, f) -> System.IO.File.ReadAllText (System.IO.Path.Combine(d, f))
+
+    let private fsFromString (source: string) (kind: SourceKind) : FSharpCompilationSource =
+        match source with
+        | null -> failwith "Source cannot be null"
+        | _ -> { Source         = Text source;
+                 Options        = defaultOptions;
+                 OutputType     = Library;
+                 SourceKind     = kind;
+                 Name           = None;
+                 IgnoreWarnings = false;
+                 References     = [] }
+
+    let private csFromString (source: string) : CSharpCompilationSource =
+        match source with
+        | null -> failwith "Source cannot be null"
+        | _ -> { Source          = Text source;
+                 LangVersion     = CSharpLanguageVersion.CSharp8;
+                 TargetFramework = TargetFramework.NetCoreApp30;
+                 Name            = None;
+                 References      = [] }
```

Comment:
Should be on new line, and the semicolons aren't required

Reply:
Semicolons is just a personal preference, will fix, thanks.